server/src/main/java/org/elasticsearch/action/admin/indices/analyze/AnalyzeResponse.java

```diff
@@ -111,6 +136,46 @@ public static AnalyzeToken readAnalyzeToken(StreamInput in) throws IOException {
             return analyzeToken;
         }
 
+        public static AnalyzeToken readAnalyzeToken(XContentParser parser) throws IOException {
+            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser::getTokenLocation);
+            String field = null;
+            String term = "";
+            int position = -1;
+            int startOffset = -1;
+            int endOffset = -1;
+            int positionLength = 1;
+            String type = "";
+            Map<String, Object> attributes = new HashMap<>();
+            for (XContentParser.Token t = parser.nextToken(); t != XContentParser.Token.END_OBJECT; t = parser.nextToken()) {
+                if (t == XContentParser.Token.FIELD_NAME) {
+                    field = parser.currentName();
+                    continue;
+                }
+                if (Fields.TOKEN.equals(field)) {
+                    term = parser.text();
+                }
+                else if (Fields.POSITION.equals(field)) {
+                    position = parser.intValue();
+                }
+                else if (Fields.START_OFFSET.equals(field)) {
+                    startOffset = parser.intValue();
+                }
+                else if (Fields.END_OFFSET.equals(field)) {
+                    endOffset = parser.intValue();
+                }
+                else if (Fields.POSITION_LENGTH.equals(field)) {
+                    positionLength = parser.intValue();
+                }
+                else if (Fields.TYPE.equals(field)) {
+                    type = parser.text();
+                }
+                else {
+                    attributes.put(field, parser.text());
+                }
+            }
+            return new AnalyzeToken(term, position, startOffset, endOffset, positionLength, type, attributes);
```

Comment:
I think `ConstructingObjectParser` or `ObjectParser` would be better here because they make it easier to be sure that the error handling is good.

Reply:
No. We honestly need one because it keeps coming up....